charts/zeppelin/templates/server/deployment.yaml

```diff
@@ -0,0 +1,159 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "zeppelin.server.deploymentName" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "zeppelin.server.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      {{- include "zeppelin.server.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "zeppelin.server.selectorLabels" . | nindent 8 }}
+      {{- with .Values.annotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      containers:
+      - name: zeppelin-server
+        image: {{ include "zeppelin.server.image" . }}
+        {{- with .Values.global.image.pullPolicy }}
+        imagePullPolicy: {{ . }}
+        {{- end }}
+        command:
+        - sh
+        - -c
+        - $(ZEPPELIN_HOME)/bin/zeppelin.sh
+        env:
+        - name: POD_UID
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.uid
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        {{- with .Values.env }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+        envFrom:
+        - configMapRef:
+            name: {{ include "zeppelin.envConfigMapName" . }}
+        volumeMounts:
```

Comment:
@Armadik Yes, you can configure the helm chart to mount a PVC to the server pod in order to persist config files like `shiro.ini` and `interpreter.json`.

Reply:
In my opinion, the shiro.ini should be mounted via ConfigMap just like the logging configuration.
Only `credentials.json`, `interpreter.json` and `notebook-authorization.json` should be outsourced. This can be done with `ZEPPELIN_CONFIG_FS_DIR`.

Other environment variables which should link to a pvc are `ZEPPELIN_NOTEBOOK_DIR` and `ZEPPELIN_SEARCH_INDEX_PATH`.

I currently use the following environment variables. The PVC is mounted under `/data`.
```
ZEPPELIN_CONFIG_FS_DIR=/data/zeppelin-config
ZEPPELIN_NOTEBOOK_DIR=/data/zeppelin-notebook
ZEPPELIN_SEARCH_INDEX_PATH=/data/lucene-index
```